hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/metrics/FlinkWriteMetrics.java

```diff
@@ -0,0 +1,130 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.metrics;
+
+import org.apache.hudi.common.model.HoodieCommitMetadata;
+import org.apache.hudi.common.table.timeline.HoodieInstantTimeGenerator;
+import org.apache.hudi.common.util.Option;
+import org.apache.hudi.common.util.collection.Pair;
+
+import org.apache.flink.metrics.MetricGroup;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.text.ParseException;
+
+/**
+ * Common flink write commit metadata metrics
+ */
+public class FlinkWriteMetrics extends HoodieFlinkMetrics {
+
+  private static final Logger LOG = LoggerFactory.getLogger(FlinkWriteMetrics.class);
+
+  protected final String actionType;
+
+  private long totalPartitionsWritten;
+  private long totalFilesInsert;
+  private long totalFilesUpdate;
+  private long totalRecordsWritten;
+  private long totalUpdateRecordsWritten;
+  private long totalInsertRecordsWritten;
+  private long totalBytesWritten;
+  private long totalScanTime;
+  private long totalCreateTime;
+  private long totalUpsertTime;
+  private long totalCompactedRecordsUpdated;
+  private long totalLogFilesCompacted;
+  private long totalLogFilesSize;
+  private long commitLatencyInMs;
+  private long commitFreshnessInMs;
+  private long commitEpochTimeInMs;
+  private long durationInMs;
+
+  public FlinkWriteMetrics(MetricGroup metricGroup, String actionType) {
+    super(metricGroup);
+    this.actionType = actionType;
+  }
+
+  @Override
+  public void registerMetrics() {
+    // register commit gauge
+    metricGroup.gauge(getMetricsName(actionType, "totalPartitionsWritten"), () -> totalPartitionsWritten);
+    metricGroup.gauge(getMetricsName(actionType, "totalFilesInsert"), () -> totalFilesInsert);
+    metricGroup.gauge(getMetricsName(actionType, "totalFilesUpdate"), () -> totalFilesUpdate);
+    metricGroup.gauge(getMetricsName(actionType, "totalRecordsWritten"), () -> totalRecordsWritten);
+    metricGroup.gauge(getMetricsName(actionType, "totalUpdateRecordsWritten"), () -> totalUpdateRecordsWritten);
+    metricGroup.gauge(getMetricsName(actionType, "totalInsertRecordsWritten"), () -> totalInsertRecordsWritten);
+    metricGroup.gauge(getMetricsName(actionType, "totalBytesWritten"), () -> totalBytesWritten);
+    metricGroup.gauge(getMetricsName(actionType, "totalScanTime"), () -> totalScanTime);
+    metricGroup.gauge(getMetricsName(actionType, "totalCreateTime"), () -> totalCreateTime);
+    metricGroup.gauge(getMetricsName(actionType, "totalUpsertTime"), () -> totalUpsertTime);
+    metricGroup.gauge(getMetricsName(actionType, "totalCompactedRecordsUpdated"), () -> totalCompactedRecordsUpdated);
```

Comment:
Do we need all these metrics? How Spark handles these metrics?

Reply:
> Do we need all these metrics?
These metrics are common commit metrics and not all are useful in compaction. However, if we can gather metrics after the write coordinator commit in the future, these metrics will be useful.  We can keep them for future extensions or add them back when needed. 
> How Spark handles these metrics? 

Spark maintain a large metrics class and will gather these metrics for all write (clustering, compaction, common writes..). We can align with spark and keep these metrics. If we add some metrics to spark commit metrics, we can migrate them to flink easily and don't need do a large diff

Reply:
> a large metrics class

Can this class be reused by Flink?

Reply:
> Can this class be reused by Flink?

Flink has its own metrics system and it seems hard to reuse hudi-spark's metrics system. For example, flink metrics core manages the `MetricsReporter` itself and we can config metrics reporters through flink config. However for hudi spark, we manager the reporter in hudi instead of Spark.  We do can reuse the metrics defined in `hudi-client-common` for flink, however in this way we will bypass Flink's metrics system which may make hudi-flink metrics different from other flink metrics. 

I think in hudi-flink, we'd better use flink's metrics system so that those who already have a metrics for flink can easily integrate hudi-flink metrics.

Reply:
Yeah, reasonable.

Reply:
I think we can make use of these commit metrics in the near future once [flink support coordinator metrics](https://cwiki.apache.org/confluence/display/FLINK/FLIP-274%3A+Introduce+metric+group+for+OperatorCoordinator). SO we can keep them now sothat we won't do large modification if we can gather metrics from `StreamwriteCoordinator`

Reply:
Can we drop these write metrics first until we introduce the coordinator metrics?

Reply:
Yes of course. Will delete them later